beacon-chain/p2p/broadcaster.go

```diff
@@ -301,13 +309,16 @@ func (s *Service) BroadcastDataColumn(ctx context.Context, columnSubnet uint64,
 	}
 
 	// Non-blocking broadcast, with attempts to discover a column subnet peer if none available.
-	go s.internalBroadcastDataColumn(ctx, columnSubnet, dataColumnSidecar, forkDigest)
+	go s.internalBroadcastDataColumn(ctx, slot, slotStartTime, root, columnSubnet, dataColumnSidecar, forkDigest)
 
 	return nil
 }
 
 func (s *Service) internalBroadcastDataColumn(
 	ctx context.Context,
+	slot primitives.Slot,
```

Comment:
Do we need the slot and slotStartTime if we can already get this information from the sidecar object

Reply:
Addressed here: 1dc8225324f4b96fc199c364889333bad998efbc

**Additional question:**
We also have the possibility to remove the `root` argument from `internalBroadcasDatacolumn`, since the hash tree root can be obtain from the data columns sidecar object with:
```go
dataColumnSidecar.SignedBlockHeader.GetHeader().HashTreeRoot()
```

However, is the cost of re-computing `HashTreeRoot` worth the removal of the `root` argument in the function?

Reply:
I think passing the root is a better idea, hashing the header 128 times will waste CPU cycles unnecessarily.